File: ts/src/images/titan/generate.ts
// takes a lot of experimentation to get the right image

import { BedrockRuntimeClient, InvokeModelCommand } from '@aws-sdk/client-bedrock-runtime'
import { writeFileSync } from 'fs'

const client = new BedrockRuntimeClient({ region: 'us-west-2' })

const titanImageConfig = {
    taskType: "TEXT_IMAGE",
    textToImageParams: {
        text: "cat on a mat on a country hillside",      
    },
    imageGenerationConfig: {
        numberOfImages: 1,
        height: 512,
        width: 512,
        cfgScale: 8.0,
    }
}

async function invokeModel() {
    const response = await client.send(new InvokeModelCommand({
        modelId: 'amazon.titan-image-generator-v1',
        body: JSON.stringify(titanImageConfig),
        accept: 'application/json',
        contentType: 'application/json'
    }));
    const responseBody = JSON.parse(new TextDecoder().decode(response.body));
    saveImage(responseBody.images[0], 'cat.png');
}

function saveImage(base64Data: string, fileName: string) {
    const imageBuffer = Buffer.from(base64Data, 'base64');
    writeFileSync(fileName, imageBuffer);
}

invokeModel();

File: py/src/images/titan/generate.py
import boto3
import json
import base64

client = boto3.client(service_name='bedrock-runtime', region_name="us-west-2")

stability_image_config = json.dumps({
    "taskType": "TEXT_IMAGE",
    "textToImageParams": {
        "text": "cat on a mat on a country hillside",      
    },
    "imageGenerationConfig": {
        "numberOfImages": 1,
        "height": 512,
        "width": 512,
        "cfgScale": 8.0,
    }
})

response = client.invoke_model(
    body=stability_image_config, 
    modelId="amazon.titan-image-generator-v1", 
    accept="application/json", 
    contentType="application/json")

response_body = json.loads(response.get("body").read())
base64_image = response_body.get("images")[0]

base_64_image = base64.b64decode(base64_image)

file_path = "cat.png"
with open(file_path, "wb") as f:
    f.write(base_64_image)
